Isosurfacing_3/include/CGAL/Isosurfacing_3/internal/Dual_contouring_internal.h

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) 2022 INRIA Sophia-Antipolis (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Daniel Zint
+//                 Julian Stahl
+
+#ifndef CGAL_DUAL_CONTOURING_3_INTERNAL_DUAL_CONTOURING_3_H
+#define CGAL_DUAL_CONTOURING_3_INTERNAL_DUAL_CONTOURING_3_H
+
+#include <CGAL/Bbox_3.h>
+#include <CGAL/Isosurfacing_3/internal/Tables.h>
+#include <CGAL/Origin.h>
+
+#include <array>
+#include <eigen3/Eigen/SVD>
+#include <map>
+#include <mutex>
+#include <vector>
+
+namespace CGAL {
+namespace Isosurfacing {
+namespace internal {
+
+namespace Positioning {
+template <bool use_bbox = false>
+class QEM_SVD {
+public:
+    /// <summary>
+    /// Compute vertex position for Dual Contouring
+    /// </summary>
+    /// <typeparam name="Domain_"></typeparam>
+    /// <param name="domain"></param>
+    /// <param name="iso_value"></param>
+    /// <param name="i"></param>
+    /// <param name="j"></param>
+    /// <param name="k"></param>
+    /// <returns> true, if there is a point in the cell</returns>
+    template <class Domain_>
+    bool position(const Domain_& domain, const typename Domain_::FT iso_value, const typename Domain_::Cell_handle& vh,
+                  typename Domain_::Point& point) const {
+        typedef typename Domain_::Point Point;
+        typedef typename Domain_::Geom_traits::Vector_3 Vector;
+        typedef typename Domain_::FT FT;
+
+        typename Domain_::Cell_vertices vertices = domain.cell_vertices(vh);
+
+        namespace Tables = internal::Cube_table;
+
+        std::array<typename Domain_::FT, Tables::N_VERTICES> s;
+        std::transform(vertices.begin(), vertices.end(), s.begin(), [&](const auto& v) { return domain.value(v); });
+
+        std::array<bool, Tables::N_VERTICES> b;
```

Comment:
@sloriot Do we also in internal files have to include all needed headers by the file seen in isolation?